src/WebJobs.Extensions/Extensions/Timers/Scheduling/ScheduleMonitor.cs

```diff
@@ -56,32 +56,57 @@ public abstract class ScheduleMonitor
                 lastStatus = new ScheduleStatus
                 {
                     Last = default(DateTime),
-                    Next = nextOccurrence
+                    Next = nextOccurrence,
+                    LastUpdated = now
                 };
                 await UpdateStatusAsync(timerName, lastStatus);
                 recordedNextOccurrence = nextOccurrence;
             }
             else
             {
                 DateTime expectedNextOccurrence;
-                if (lastStatus.Last == default(DateTime))
+
+                // Track the time that was used to create 'expectedNextOccurrence'.
+                DateTime lastUpdated;
+
+                if (lastStatus.LastUpdated != default(DateTime))
```

Comment:
nit: could collapse this whole if else to something like:

```
DateTime lastUpdated = lastStatus?.LastUpdated ?? lastStatus?.Last ?? now;
expectedNextOccurrence = schedule.GetNextOccurrence(lastUpdated);
```

I like the comments, def helps the grok.